main.go

```diff
@@ -84,74 +81,50 @@ func DownloadFile(downloadURL, targetPath string) error {
 	return nil
 }
 
-func main() {
-	configs := createConfigsModelFromEnvs()
-
+// downloadNuGet downloads NuGet with the given version.
+func downloadNuGet(version string) (string, error) {
 	fmt.Println()
-	configs.print()
-
-	if err := configs.validate(); err != nil {
-		log.Errorf("Issue with input: %s", err)
-		os.Exit(1)
+	log.Infof("Downloading NuGet %s version...", version)
+	tmpDir, err := pathutil.NormalizedOSTempDirPath("__nuget__")
+	if err != nil {
+		return "", fmt.Errorf("failed to create tmp dir, error: %s", err)
 	}
 
-	nugetPth := "/Library/Frameworks/Mono.framework/Versions/Current/bin/nuget"
-	nugetRestoreCmdArgs := []string{nugetPth}
+	downloadPth := filepath.Join(tmpDir, "nuget.exe")
 
-	if configs.NugetVersion != "" {
-		fmt.Println()
-		log.Infof("Downloading Nuget %s version...", configs.NugetVersion)
-		tmpDir, err := pathutil.NormalizedOSTempDirPath("__nuget__")
-		if err != nil {
-			log.Errorf("Failed to create tmp dir, error: %s", err)
-			os.Exit(1)
-		}
+	// https://dist.nuget.org/win-x86-commandline/latest/nuget.exe or
+	// https://dist.nuget.org/win-x86-commandline/v3.3.0/nuget.exe
 
-		downloadPth := filepath.Join(tmpDir, "nuget.exe")
+	if version != "latest" {
+		version = `v` + version
+	}
+	nuGetURL := fmt.Sprintf("https://dist.nuget.org/win-x86-commandline/%s/nuget.exe", version)
 
-		// https://dist.nuget.org/win-x86-commandline/latest/nuget.exe or
-		// https://dist.nuget.org/win-x86-commandline/v3.3.0/nuget.exe
-		version := configs.NugetVersion
-		if version != "latest" {
-			version = `v` + version
+	log.Printf("Download URL: %s", nuGetURL)
+	return downloadPth, retry.Times(1).Wait(time.Second).Try(func(attempt uint) error {
+		if attempt > 0 {
+			log.Warnf("Retrying...")
 		}
-		nugetURL := fmt.Sprintf("https://dist.nuget.org/win-x86-commandline/%s/nuget.exe", version)
-
-		log.Printf("Download URL: %s", nugetURL)
-
-		if err := retry.Times(1).Wait(time.Second).Try(func(attempt uint) error {
-			if attempt > 0 {
-				log.Warnf("Retrying...")
-			}
-			if err := DownloadFile(nugetURL, downloadPth); err != nil {
-				log.Errorf("Failed to download nuget, error: %s", err)
-				return err
-			}
-			return nil
-		}); err != nil {
-			log.Errorf("Failed to download nuget, error: %s", err)
-			os.Exit(1)
+		if err := DownloadFile(nuGetURL, downloadPth); err != nil {
+			log.Errorf("Failed to download NuGet, error: %s", err)
+			return err
 		}
+		return nil
+	})
+}
 
-		nugetRestoreCmdArgs = []string{constants.MonoPath, downloadPth}
-	}
-
-	fmt.Println()
-	log.Infof("Restoring Nuget packages...")
-
-	nugetRestoreCmdArgs = append(nugetRestoreCmdArgs, "restore", configs.XamarinSolution)
-
-	if err := retry.Times(1).Try(func(attempt uint) error {
+// runRestoreCommand runs the restore command with the given args.
+func runRestoreCommand(nuGetRestoreCmdArgs []string) error {
```

Comment:
NOTE: nuGetRestoreCmdArgs could be way shorter as the scope of the variable is short, something like `cmd []string`, with this change the command.Model variable needs to be renamed for `c` or similar.

Reply:
Changed it to cmdArgs, so the other variable could remain cmd.